runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/graph/QueryablePipeline.java

```diff
@@ -239,16 +240,16 @@ public Components getComponents() {
    * Returns the {@link PCollectionNode PCollectionNodes} that the provided transform consumes as
    * side inputs.
    */
-  public Collection<PCollectionNode> getSideInputs(PTransformNode transform) {
+  public Map<String, PCollectionNode> getSideInputs(PTransformNode transform) {
     return getLocalSideInputNames(transform.getTransform())
         .stream()
         .map(
             localName -> {
               String pcollectionId = transform.getTransform().getInputsOrThrow(localName);
-              return PipelineNode.pCollection(
-                  pcollectionId, components.getPcollectionsOrThrow(pcollectionId));
+              return KV.of(localName, PipelineNode.pCollection(
```

Comment:
It's a little weird to use the Beam KV type here rather than Map.Entry, but it's unfortunate that Java doesn't provide a built in Pair type. Not saying you should change this here, but I wanted to point it out.